cli/src/ui.rs

```diff
@@ -314,6 +392,11 @@ impl Ui {
     /// Waits for the pager exits.
     #[instrument(skip_all)]
     pub fn finalize_pager(&mut self) {
+        if let UiOutput::Builtin { pager } =
+            mem::replace(&mut self.output, UiOutput::new_terminal())
+        {
+            pager.finalize();
+        }
```

Comment:
better to use `match mem::replace(&mut self.output, UiOutput::new_terminal()) {` because doing that twice doesn't make sense.